Move some ToString() functions to IRac.cpp

crankyoldgit opened the pull request:
* Move some `ToString()` functions to IRac.cpp
* Add unit test coverage for these functions.
* Create a IRSTRING define/type to simplify Arduino vs Std c++ string handling.

Fixes #747

crankyoldgit committed:
Move IRac to String functions to IRac.cpp
* Move IRac to String functions to IRac.cpp
* Add unit test coverage for these functions.
* Create a IRSTRING define/type to simplify Arduino vs Std c++ string handling.

Fixes #747

jimmys01 commented on examples/IRMQTTServer/IRMQTTServer.ino:
```diff
@@ -3058,17 +2974,17 @@ bool sendClimate(const stdAc::state_t prev, const stdAc::state_t next,
     diff = true;
     success &= sendString(topic_prefix + KEY_SWINGV,
-                          swingvToString(next.swingv), retain);
+                          IRac::swingvToString(next.swingv), retain);
```
May i ask why IRac:: is needed?

crankyoldgit replied:
They are a [static member function](https://www.tutorialspoint.com/cplusplus/cpp_static_members.htm) of the `IRac` class now after the move. The class doesn't need to be instantiated, so it shouldn't take up extra memory/prog space. Plus it keeps the namespace a bit cleaner.

Basically, they are standalone functions that don't need the class object, but are associated with it.

crankyoldgit committed:
Make Arduino `String` type available in UNIT_TEST
- Remove a whole bunch of `UNIT_TEST` defines because `String` is now available as a typedef.
- Remove redundant code.
- Add some missing `String.reserve()`s to some functions.

Ref #747

crankyoldgit committed:
Merge branch 'master' into Issue747
